frontend/src/app/shared/toggle/toggle.component.html

```diff
@@ -0,0 +1,4 @@
+<div class="toggle-container">
+    <input type="checkbox" (click)="toggleButton()"/>
+    <label class="slider"></label>
+</div>
```

Comment:
This input is missing a an accessible label (or the label is missing text?)

Reply:
The `<label>` element is here purely to be able to attach pseudo-elements to it from css. Adding text will break the toggle button. The description of what the toggle button does should come from the component in which it is used; the "on / off" state is reflected by screen readers as it is built on top of an `<input>` element.

Reply:
Ah, I see. In that case:

- This element is not an actual label, so the semantics are misleading. If it's just for CSS, you can use a generic element instead (`<div>` or `<span>`). If that is not possible, it should set `role="generic"`.
- While the _value_ of the checkbox is machine-readable, the _label_ is not. That association exists visually, but will not show in the accessibility tree. For screen reader users, this creates the issue that it may be unclear what the checkbox is meant to do. Unlabelled form elements also make it harder to navigate a page using a screen reader (as you could otherwise use the overview of form elements).

The most typical technique for providing accessible labels is [using a `<label>` with "for" attribute](https://www.w3.org/WAI/WCAG22/Techniques/html/H44.html) but the component hierarchy may get in the way here. You could embed the input inside the `<label>` elements; see some examples in [MDN's documentation on `<label>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label). (I'm not sure if that also works when the input is not a direct child of the label, though.) This technique also has the advantage of increasing the clickable area.

An alternative technique would be to [use an `aria-labelled-by` attribute](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA16.html).